apps/web/src/lib/components/book-reader/book-reader-paginated/page-manager-paginated.ts

```diff
@@ -174,4 +196,33 @@ export class PageManagerPaginated implements PageManager {
       return subscription;
     });
   }
+
+  private updateSectionData(ref: string, progress: number, emit = true) {
+    if (ref && this.sectionData.has(ref)) {
```

Comment:
For such code I usually prefer early return, like
`if (!ref || !this.sectionData.has(ref)) return`